build(edr_napi): migrate to napi-rs v3 (#1385)
NomicFoundation/edr: .changeset/silver-baths-fold.md

---
"@nomicfoundation/edr": minor
---

> - Migrated `edr_napi` to napi-rs v3.

Reviewer: Suggestion: I think the below changes already cover the user-facing API changes. I think the migration itself is an internal detail.
```suggestion
```

- The `reason`, `counterexample`, and `valueSnapshotGroups` fields on `TestResult` are now class getters returning `T | undefined`.

> - `SuiteResult` is now a plain object instead of a class; field shapes are unchanged.

Reviewer: Style: conform to changelog format
```suggestion
- Changed the `reason`, `counterexample`, and `valueSnapshotGroups` fields on `TestResult` to class getters returning `T | undefined`.
- Changed `SuiteResult` from a class to a plain object; field shapes are unchanged.
```

> - Exceptions thrown by the `decodeConsoleLogInputsCallback` and `printLineCallback` logger callbacks now surface as JSON-RPC internal-error responses carrying the JS error message, instead of being swallowed or crashing the process.

Reviewer: ```suggestion
- Fixed exceptions thrown by the `decodeConsoleLogInputsCallback` and `printLineCallback` logger callbacks from being swallowed or crashing the process. They now surface as JSON-RPC internal-error responses carrying the JS error message.
```
